src/script/interpreter.cpp

```diff
@@ -1937,6 +1995,58 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                 }
                 break;
 
+                case OP_ECMULSCALAREXPVERIFY:
+                {
+                    // OP_ECMULSCALAREXPVERIFY is available post tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
+                    valtype& vchScalar = stacktop(-1);
+                    valtype& vchGenerator = stacktop(-2);
+                    valtype& vchRes = stacktop(-3);
+
+                    CPubKey pk(vchGenerator);
+                    CPubKey res(vchRes);
+                    if (vchGenerator.size() != CPubKey::COMPRESSED_SIZE || vchRes.size() != CPubKey::COMPRESSED_SIZE)
```

Comment:
We can more directly check with `!pk.IsCompressed() || !res.IsCompressed()`.

Can you also add a test with an invalid public key (one that is a properly encoded compressed key, just not on the curve) and one with a bad encoding? I'm not sure if you need to do an `IsValid` or `IsFullyValid` check here, or if these are covered by `pk.TweakMulVerifiy`.

Reply:
I was operating under the assumption that `scep` API should take care of all invalid inputs and the caller should not worry about these things. I will double-check.

Reply:
It does do the check. Adding tests for checking this.